### DISCUSSION TITLE
How to run default resolvers in parallel?

### DISCUSSION THREAD

**Discussion by xsiddiqmx1** [mirumee/ariadne#1085] [Q&A]
My root resolver returns the complete response and then the default resolvers are then used to fetch the data. How can I make the default resolvers fetch the data in parallel instead of sequentially?

**Comment by rafalp** [comment]
Hello, I have some problems understanding your setup. If your root resolver returns complete response, then why do you fetch the data in default resolvers?

> How can I make the default resolvers fetch the data in parallel instead of sequentially?

If you make your resolvers async, they will be called and awaited in `gather` (but not for fields on `Mutation` type, as per spec), but if there's no IO being executed, this will be slower in Python than sequential execution.

**Reply by xsiddiqmx1** [reply]
My root resolver returns fully populated obj which my schema supports. Graphql then would use default resolvers to get the data based on the request. I have used Graphql Java and that's how I understood it, default getters were called to formulate the response.

**Reply by dacevedo12** [reply]
If you're using asyncio, graphql-core is already resolving them concurrently (with `asyncio.gather`)

The sequential step of resolving the parent and then its children remains. If that is what you meant, the only way would be to implement some sort of lookahead

Would be really useful if you are able to provide a short example of what you're trying to achieve

**Reply by xsiddiqmx1** [reply]
Sure! 
I am using graphql_sync(schema,data,context_value=event,logger=logger)
Sample Query:
query ($input: String) { places(input: $input) { name country code { id desc}}}
Schema:
 """
   type Query {
       places(input:String): [Place]
   }
   type Place {
       name: String!
       description: String!
       country: String!
       code: Code
       }
   type Code {
       id: String
       desc: String
   }
   """

#Response Code
@query_type.field("places")
def places(obj, info, input):
    return [
        {"name": "Paris", "description": "The city of lights", "country": "France",
         "code": {"id": "750922", "desc": "zip"}},
        {"name": "Rome", "description": "The city of pizza", "country": "Italy",
         "code": {"id": "750922", "desc": "zip"}},
        {
            "name": "London",
            "description": "The city of big buildings",
            "country": "United Kingdom",
        },
    ]
I am trying to understand after the "places" function returns the list does it populate the response in the following sequence
name-> country->code->id -> desc.
Anyway, for it to set the name, and country in parallel instead of sequentially. 
In the actual application, the schema is large and the response will have hundreds of rows so trying to see if things can run in parallel.
Hope my example makes sense.
